java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java

```diff
@@ -253,28 +255,25 @@ public static void subscribeMinionToChannels(SystemQuery systemQuery, MinionServ
                         // ActivationKey without base channel (SUSE Manager Default)
                         () -> {
                             Set<SUSEProduct> suseProducts = identifyProduct(systemQuery, server, grains);
-                            return findChannelsForProducts(suseProducts, minionId);
+                            Set<Channel> channelsForProducts = findChannelsForProducts(suseProducts, minionId);
+                            Optional<Channel> baseChannel = channelsForProducts.stream()
+                                    .filter(c -> c.isBaseChannel()).findFirst();
+                            if (baseChannel.isPresent()) {
+                                // assign the identified base channel and activation key channels related
+                                // to that base channel only
+                                return Stream.concat(
+                                        Stream.of(baseChannel.get()),
+                                        ak.getChannels().stream()
+                                                .filter(c -> c.getParentChannel().getId()
+                                                        .equals(baseChannel.get().getId())))
+                                        .collect(toSet());
+                            }
+                            else {
+                                return emptySet();
+                            }
                         },
-                        baseChannel -> Opt.fold(
-                                SUSEProductFactory.findProductByChannelLabel(baseChannel.getLabel()),
-                                () -> {
-                                    // ActivationKey with custom channel
-                                    return Stream.concat(
-                                            Stream.of(baseChannel),
-                                            ak.getChannels().stream()
-                                    ).collect(toSet());
-                                },
-                                baseProduct -> {
-                                    // ActivationKey with vendor or cloned vendor channel
-                                    return Stream.concat(
-                                            lookupRequiredChannelsForProduct(baseProduct),
-                                            ak.getChannels().stream()
-                                                    .filter(c -> c.getParentChannel() != null &&
-                                                            c.getParentChannel().getId().equals(baseChannel.getId()))
-                                    ).collect(toSet());
-
-                                }
-                        )
+                        // assign base channel and activation key channels selected only
+                        baseChannel -> Stream.concat(Stream.of(baseChannel), ak.getChannels().stream()).collect(toSet())
```

Comment:
Is our expectation here, that a valid and sufficient channel set is provided? Or may it happen that mandatory channels are missing?

Reply:
Mandatory channels should never be missing, but maybe double checking and adding them here could be a good option to stay on the safe side. For sure, we should not add recommended ones anyway, even if the system will not be "really usable" in the end. If a user uses an AK it's because he wants to manage the selection, if he wants something with our magic knowledge either he uses the toggler for recommended or he does not use AK at all. I can understand we are trying to be nice, but it is tricky to understand, and if a user really wants to customize a selection, he will be puzzled and not able to customize it for real.

Reply:
I think mainly about XMLRPC. Not sure what checks we have there to "autoselect" dependend  channels.
I agree to only check for mandatory channels - not recommended.